packages/jest-haste-map/src/crawlers/node.js

```diff
@@ -127,20 +127,28 @@ function findNative(
 module.exports = function nodeCrawl(
   options: CrawlerOptions,
 ): Promise<InternalHasteMap> {
-  const {data, extensions, forceNodeFilesystemAPI, ignore, roots} = options;
+  const {
+    data,
+    extensions,
+    forceNodeFilesystemAPI,
+    ignore,
+    rootDir,
+    roots,
+  } = options;
 
   return new Promise(resolve => {
     const callback = list => {
       const files = new Map();
       list.forEach(fileData => {
-        const name = fileData[0];
+        const filePath = fileData[0];
+        const relativeFilePath = path.relative(rootDir, filePath);
```

Comment:
should we do any OS normalization? If the point of this is remote caching, this won't work between posix and windows, will it?

Reply:
The way it's implemented now it wouldn't work between posix and windows. I guess we should normalize the paths, if I can do it without significant perf impact

Reply:
Just doing `path.posix.relative` instead of `path.relative` should work, not sure if it would actually break in edge-cases for windows, though

Reply:
> Just doing path.posix.relative instead of path.relative should work

Yes but you'd have to normalize the path every time you access the haste map, which seems too much.

I considered making the paths relative only during serialization (which would've made this easier for windows) but that seriously impacts startup time. Doing it in runtime, as it is done now, removes that impact but requires doing normalization in runtime, which makes it harder to maintain compatibility between posix and windows. This is a tricky one and I'm not sure it's worth the effort.

Maybe we should only support remote caching when the haste map is created using the same path format.

Reply:
More on that, since it'll be an undocumented feature I think we can go with this version for now. If someone needs it in the future we can revisit it.

Reply:
Yeah, sounds good!👍